### PR TITLE
Closes #206 - Oracle client cache should use guava

### LINKED ISSUE #206: OracleClient should be retrieved from Environment
There's a cache of client objects in `OracleClient` that should use a guava instead of just a plain old map (maybe, I don't feel very strongly about).

### PR DESCRIPTION
**Pull request by mikewalch**
PR for #206

### CODE REVIEW

File: modules/core/src/main/java/io/fluo/core/oracle/OracleClient.java [hunk 1]
```diff
@@ -270,8 +272,10 @@ public void takeLeadership(CuratorFramework curatorFramework) throws Exception {
     }
   }
 
-  private final static Map<String,OracleClient> clients = new HashMap<>();
-
+  private final static Cache<String, OracleClient> clients = CacheBuilder.newBuilder()
```
Comment (keith-turner) [inline modules/core/src/main/java/io/fluo/core/oracle/OracleClient.java]: Wondering if we even need this static cache and `getInstance()` method.    May be better if an Oracle client could be obtained from an `Environment ` object and the cache dropped.  I suspect this cache predates using the Environment class to share resources.

Comment (mikewalch) [inline modules/core/src/main/java/io/fluo/core/oracle/OracleClient.java]: Sounds good to me.  Fixed in 6ff45ca447d3fbc4

File: modules/core/src/main/java/io/fluo/core/impl/SharedResources.java [hunk 2]
```diff
@@ -79,6 +81,14 @@ public CuratorFramework getCurator() {
     return curator;
   }
   
+  public synchronized OracleClient getOracleClient() {
+    checkIfClosed();
+    if (oracleClient == null) {
+      oracleClient = new OracleClient(env);
+    }
+    return oracleClient;
```
Comment (keith-turner) [inline modules/core/src/main/java/io/fluo/core/impl/SharedResources.java]: Now that shared resources is tracking it, we should probably close it.  I was looking at the code for OracleClient and I think I know what needs to be done to add a close method (also I have worked on this code w/ @cjnolet  before).  I am thinking of making pull request against your branch to add that.  Is that ok or would you rather work on adding the close method?

Comment (mikewalch) [inline modules/core/src/main/java/io/fluo/core/impl/SharedResources.java]: I was thinking that OracleClient probably needed a close method but I am not too familiar with it.  Feel free to make a pull request.

Comment (keith-turner) [inline modules/core/src/main/java/io/fluo/core/impl/SharedResources.java]: ok made PR mikewalch/fluo#1

**Comment by keith-turner** [comment]
+1

**Pull request merged**